RecoLocalCalo/HcalRecProducers/src/HBHEPhase1Reconstructor.cc

```diff
@@ -121,6 +127,48 @@ namespace {
     {
         return std::pair<bool,bool>(df.linkError(), df.capidError());
     }
+
+    std::unique_ptr<HBHEStatusBitSetter> parse_HBHEStatusBitSetter(
+        const edm::ParameterSet psdigi)
+    {
+        return std::make_unique<HBHEStatusBitSetter>(
+            psdigi.getParameter<double>("nominalPedestal"),
+            psdigi.getParameter<double>("hitEnergyMinimum"),
+            psdigi.getParameter<int>("hitMultiplicityThreshold"),
+            psdigi.getParameter<std::vector<edm::ParameterSet> >("pulseShapeParameterSets"));
+    }
+
+    std::unique_ptr<HBHEPulseShapeFlagSetter> parse_HBHEPulseShapeFlagSetter(
+        const edm::ParameterSet &psPulseShape)
+    {
+        return std::make_unique<HBHEPulseShapeFlagSetter>(
+            psPulseShape.getParameter<double>("MinimumChargeThreshold"),
+            psPulseShape.getParameter<double>("TS4TS5ChargeThreshold"),
```

Comment:
as another refactoring item for the future,
may I ask to replace function calls with  N>~5 arguments by calls using a struct. So that each argument is easily traceable when reading the code or debugging, or when the list needs an update to remove/add/rename an argument in the future developments.